gateway/gateway.py

```diff
@@ -269,11 +335,81 @@ def __init__(self, mqtt_prefix, bolt_sensor=False):
 }
 api_users = {}
 
+
+def door_from_path(path):
+    # /doors/virtual-1/unlock?...
+    if not path.startswith('/doors'):
+        return None
+
+    tok = path.split('/')
+    door = tok[2]
+    return door
+
+def door_from_topic(topic):
+    # doors/device-100/unlock
+    #m = re.match(r"doors\/(.*)\/.*")
+    tok = topic.split('/')
+    if len(tok) <= 2:
+        return None
+    prefix = '/'.join(tok[0:2])
+    door = door_id_from_mqtt(prefix)
+    return door
+
+def request_log_params():
+    r, g = flask.request, flask.g
+
+    user = None if not r.authorization else r.authorization.username
+    query = f'{r.path}?{r.query_string.decode("utf-8")}'
+    door = door_from_path(r.path)
+
+    log_params = dict(
+        request_id=g.request_id,
+        method=r.method,
+        path=query,
+        user=user,
+        door=door,
+    )
+    return log_params
+
+
+## Logging helpers
+def log_request(sender, **extra):
+    r, g = flask.request, flask.g
+
+    # add info
+    g.request_id = r.headers.get('X-Request-Id', str(uuid.uuid4()))
+    g.request_start_time = time.time()
+
+    log_params = request_log_params()
+    log_params.update(dict(
+    ))
+
+    log.info('http-request-start', **log_params)
+
+def log_response(sender, response, **extra):
+    r, g = flask.request, flask.g
+
+    g.request_end_time = time.time()
+    duration = 1000.0 * (flask.g.request_end_time - flask.g.request_start_time)
+
+    log_params = request_log_params()
+    log_params.update(dict(
+        status_code=response.status_code,
+        duration_ms=duration,
+    ))
+
+    log.info('http-request-end', **log_params)
```

Comment:
If these could push at least the request path to a thread local dict that would be useful. makes it easier to correlate stuff. https://www.structlog.org/en/stable/thread-local.html

If it could add a "request_id" field which is just an uuid we can trace all logging from a specific request which is also nice.

Reply:
request path and request_id is included, see the request_log_param() helper. Though not all logged events have it yet. Will try to use the threadlocal mechanism instead of doing it explicitly, to get it everywhere